### PR TITLE
Jnode fs xfs changes

### PR DESCRIPTION
**Pull request by JulioCesarParra**
Hi, Erwin, I have implemented the code changes to solve the 3 issues that were open to the XFS feature.

### CODE REVIEW

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public static void buildXfsMetaDataStructure(FSEntry entry, StringBuilder actual
      * @param indent the indent level.
      *
      */
-    private static StringBuilder getXfsMetadata(FSEntry entry, StringBuilder actual,String indent) throws IOException {
+    private static StringBuilder getXfsMetadata(XfsEntry entry, StringBuilder actual,String indent) throws IOException {
         actual.append(indent);
         actual.append(indent);
         actual.append("atime : " +  getDate(((FSEntryLastAccessed) entry).getLastAccessed() ));
```
Comment (CoffeeForOven) [inline src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java]: You should no longer need to cast entry on this line and the next, as `XfsEntry` implement these interfaces.

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java [hunk 2]
```diff
@@ -180,10 +180,10 @@ public void testCentos() throws Exception {
         try (FileDevice device = new FileDevice(testFile, "r")) {
             XfsFileSystemType type = new XfsFileSystemType();
             XfsFileSystem fs = type.create(device, true);
-            FSEntry entry = fs.getRootEntry();
+            XfsEntry entry = fs.getRootEntry();
```
Comment (CoffeeForOven) [inline src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java:183]: I would've thought that the return value from `fs.getRootEntry()` would needed to be cast to `XfsEntry` like it is in `testXfsMetaData()`.

Comment (CoffeeForOven) [reply src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java:183]: Never mind. I can see why it doesn't need to be cast.

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java [hunk 3]
```diff
@@ -180,10 +180,10 @@ public void testCentos() throws Exception {
...
             StringBuilder actual = new StringBuilder(1024);
             buildXfsDirStructure(entry, actual, "  ");
-            System.out.println(actual);
+            //System.out.println(actual);
```
Comment (CoffeeForOven) [inline src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java]: Might as well remove commented out code.

File: src/main/java/org/jnode/fs/xfs/XfsEntry.java [hunk 4]
```diff
@@ -30,7 +30,10 @@
      */
     private static final Logger log = LoggerFactory.getLogger(BlockDirectoryEntry.class);
 
-    long NANO_SECOND = 1000000000;
+    /**
+     * Constant used to convert nano seconds to milliseconds.
+     */
+    private static final long ONE_MILLISECOND = 1_000_000;
```
Comment (CoffeeForOven) [inline src/main/java/org/jnode/fs/xfs/XfsEntry.java]: The constant name isn't quite right. It's strange to say that `ONE_MILLISECOND` is 1,000,000. Suggestion below.

File: src/main/java/org/jnode/fs/xfs/XfsEntry.java [hunk 5]
```diff
@@ -73,19 +76,20 @@ public String getId() {
...
     @Override
     public long getLastAccessed() throws IOException {
-        return (inode.getAccessTimeSec() * 1000) + (inode.getAccessTimeNsec() / NANO_SECOND);
+        return (inode.getAccessTimeSec() * 1000) + (inode.getAccessTimeNsec() / ONE_MILLISECOND);
     }
 
     @Override
     public long getLastChanged() throws IOException {
-        return (inode.getChangedTimeSec() * 1000) + (inode.getChangedTimeNsec() / NANO_SECOND);
+        return (inode.getChangedTimeSec() * 1000) + (inode.getChangedTimeNsec() / ONE_MILLISECOND);
```
Comment (CoffeeForOven) [inline src/main/java/org/jnode/fs/xfs/XfsEntry.java]:
```suggestion
    /**
     * Converts an entry's time value from the seconds/nanoseconds values to a millisecond
     * value, usable by {@link #getCreated()}, {@link #getLastAccessed()}, and {@link #getLastChanged()}.
     *
     * @param seconds     the seconds value from the entry.
     * @param nanoseconds the nanoseconds value from the entry.
     * @return the milliseconds value.
     * @see #getCreated()
     * @see #getLastAccessed()
     * @see #getLastChanged()
     */
    private long getMilliseconds(long seconds, long nanoseconds) {
        return (seconds * 1_000) + (nanoseconds / 1_000_000);
    }

    @Override
    public long getCreated() {
        return getMilliseconds(inode.getCreatedTimeSec(), inode.getCreatedTimeNsec());
    }

    @Override
    public long getLastAccessed() {
        return getMilliseconds(inode.getAccessTimeSec(), inode.getAccessTimeNsec());
    }

    @Override
    public long getLastChanged() {
        return getMilliseconds(inode.getChangedTimeSec(), inode.getChangedTimeNsec());

```

Comment (JulioCesarParra) [reply src/main/java/org/jnode/fs/xfs/XfsEntry.java]: HI , Erwin your suggestion was applied . Thanks

File: src/main/java/org/jnode/fs/xfs/XfsEntry.java (hunk 4 above)
Comment (JulioCesarParra) [reply src/main/java/org/jnode/fs/xfs/XfsEntry.java]: HI , Erwin your suggestion was applied . Thanks

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java (hunk 3 above)
Comment (JulioCesarParra) [reply src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java]: HI , Erwin your suggestion was applied . Thanks

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java (hunk 2 above)
Comment (JulioCesarParra) [reply src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java:183]: HI , Erwin, was not necessary to cast to XfsEntry. Thanks

File: src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java (hunk 1 above)
Comment (JulioCesarParra) [reply src/test/java/org/jnode/fs/xfs/XfsFileSystemTest.java]: HI , Erwin your suggestion was applied . Thanks

**Pull request merged**